shared/subspace-data-retrieval/src/object_fetcher.rs

```diff
@@ -238,11 +241,19 @@ where
     /// putting the objects' bytes together.
     ///
     /// Checks the objects' hashes to make sure the correct bytes are returned.
+    ///
+    /// For efficiency, objects in a batch should be sorted by increasing piece index. Objects with
+    /// the same piece index should be sorted by increasing offset. This allows the last piece to
+    /// be re-used for the next object in the batch.
+    ///
+    /// Batches should be split if the gap between object piece indexes is 6 or more. Those objects
+    /// can't share any pieces, because a maximum-sized object only uses 6 pieces.
```

Comment:
Do we need to duplicate the comment here and Api ?
Just here would be ideal since API does not need to know the inner working on the actual fetch_objects code path

Reply:
Yes, because users of the RPC and Rust APIs both need to know how to send batches to get the best performance from the code.